### ISSUE TITLE
Reopen Issue #48: Maybe unavailable for table field name with Quote Character

### ISSUE THREAD

**Issue by tangyibo** [bytefish/PgBulkInsert#53]
Hello @bytefish :
     Recently, I found issue #48 is wrong in code file : de.bytefish.pgbulkinsert.row.SimpleRowWriter.java, function: getCopyCommand().
    And I make the follow test:

```
import java.util.Arrays;
import java.util.stream.Collectors;

public class CollectionStreamTester {

	public static final char QuoteChar = '"';

	private static boolean requiresQuoting(String identifier) {

		char first = identifier.charAt(0);
		char last = identifier.charAt(identifier.length() - 1);

		if (first == QuoteChar && last == QuoteChar) {
			return false;
		}

		if (!Character.isLowerCase(first) && first != '_') {
			return true;
		}

		for (int i = 1; i < identifier.length(); i++) {
			char c = identifier.charAt(i);

			if (Character.isLowerCase(c)) {
				continue;
			}

			switch (c) {
			case '0':
			case '1':
			case '2':
			case '3':
			case '4':
			case '5':
			case '6':
			case '7':
			case '8':
			case '9':
			case '_':
			case '$': // yes it's true
				continue;
			}

			return true;
		}

		return false;
	}

	public static String quoteIdentifier(String identifier) {
		return requiresQuoting(identifier) ? (QuoteChar + identifier + QuoteChar) : identifier;
	}

	public static void main(String[] args) {
		boolean usePostgresQuoting = true;
		String tableColumns[] = { "id", "name", "binary" };

		String commaSeparatedColumns = Arrays.stream(tableColumns)
				.map(x -> usePostgresQuoting ? CollectionStreamTester.quoteIdentifier(x) : x)
				.collect(Collectors.joining(", "));
		System.out.println(commaSeparatedColumns);
	}

}
```

The program print result ：
```
id, name, binary
```
but not is ：
```
"id", "name", "binary"
```

so I thing maybe there is something wrong in the stream function code ：

```
    private static String getCopyCommand(Table table, boolean usePostgresQuoting) {

        String commaSeparatedColumns = Arrays.stream(table.columns)
                .map(x -> usePostgresQuoting ? PostgreSqlUtils.quoteIdentifier(x) : x)
                .collect(Collectors.joining(", "));

        return String.format("COPY %1$s(%2$s) FROM STDIN BINARY",
                table.GetFullyQualifiedTableName(usePostgresQuoting),
                commaSeparatedColumns);
    }
```

Can you help me? thanks!!
```

**Comment by bytefish** [comment]
Well, yes. I don't maintain a list of reserved keywords, which might be the issue here. Generally lower-case column names don't need quoting, that's why these columns aren't quoted in your example.

I don't know, what could happen if we just quote everything, when UsePostgresQuoting is set to true?

Do you know of any problems, this might introduce?

**Comment by tangyibo** [comment]
Well, yes , i find , bad grammer will happen when some tables that have field name using SQL keyword like 'binary' or 'BINARY', so I think it is safe to quote table name and field name, Do you aggree with me? Thanks!

**Comment by tangyibo** [comment]
Thanks, I find you already fix it!

**Comment by bytefish** [comment]
I wrote a comment to say I fixed it, but I probably didn't send it. Sorry. Yes I have released 4.1 and now simply quote everything, which isn't already quoted, if you choose Postgres Quoting.
